Tests/OpenCombineTests/PublisherTests/MapTests.swift

```diff
@@ -0,0 +1,156 @@
+        // Given
+        let publisher = PassthroughSubject<Int, TestingError>()
+        let map = publisher.map { $0 * 2 }
+        let tracking = TrackingSubscriber(receiveSubscription: { $0.request(.unlimited) })
+        // When
+        publisher.send(1)
+        map.subscribe(tracking)
+        publisher.send(2)
+        publisher.send(3)
+        publisher.send(completion: .finished)
+        publisher.send(5)
+        // Then
+        XCTAssertEqual(tracking.history, [
+            .subscription(Subscriptions.empty),
+            .value(4),
+            .value(6),
+            .completion(.finished)
+        ])
+    }
+
+    func testDemandSubscribe() {
+        // Given
+        let expectedSubscribeDemand = 42
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let map = publisher.map { $0 * 2 }
+        let tracking = TrackingSubscriber(receiveSubscription: { $0.request(.max(expectedSubscribeDemand)) })
+        // When
+        map.subscribe(tracking)
+        // Then
+        XCTAssertEqual(subscription.history, [.requested(.max(expectedSubscribeDemand))])
+    }
+
+    func testDemandSend() {
+        // Given
+        let expectedReceiveValueDemand = 4
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let map = publisher.map { $0 * 2 }
+        let tracking = TrackingSubscriber(receiveValue: { _ in .max(expectedReceiveValueDemand) })
+        // When
+        map.subscribe(tracking)
+        // Then
+        XCTAssertEqual(publisher.send(0), .max(expectedReceiveValueDemand))
+    }
+
+    func testCompletion() {
+        // Given
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let map = publisher.map { $0 * 2 }
+        let tracking = TrackingSubscriber(receiveSubscription: { $0.request(.unlimited) })
+        // When
+        map.subscribe(tracking)
+        publisher.send(completion: .finished)
+        // Then
+        XCTAssertEqual(subscription.history, [.requested(.unlimited)])
+        XCTAssertEqual(
+            tracking.history,
+            [.subscription(Subscriptions.empty),.completion(.finished)]
+        )
+    }
+
+    func testCancel() throws {
+        // Given
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let map = publisher.map { $0 * 2 }
+        var downstreamSubscription: Subscription?
+        let tracking = TrackingSubscriber(receiveSubscription: {
+            $0.request(.unlimited)
+            downstreamSubscription = $0
+        })
+        // When
+        map.subscribe(tracking)
+        try XCTUnwrap(downstreamSubscription).cancel()
+        // Then
+        XCTAssertEqual(subscription.history, [.requested(.unlimited), .canceled])
+    }
+
+    func testCancelAlreadyCancelled() throws {
```

Comment:
This test's name is slightly misleading, I don't see that the subscription is canceled twice. It makes sense to either update the name of the test (having in mind that `allTests` should also be updated), or add one more test for actual double-cancel.

Reply:
Added second Cancellation, that was the idea of test